chromium_src/components/permissions/permission_context_base.cc

```diff
@@ -44,4 +59,75 @@ void PermissionContextBase::PermissionDecided(
       content_setting, is_one_time);
 }
 
+void PermissionContextBase::DecidePermission(
+    content::WebContents* web_contents,
+    const PermissionRequestID& id,
+    const GURL& requesting_origin,
+    const GURL& embedding_origin,
+    bool user_gesture,
+    BrowserPermissionCallback callback) {
+  PermissionContextBase_ChromiumImpl::DecidePermission(
+      web_contents, id, requesting_origin, embedding_origin, user_gesture,
+      std::move(callback));
+
+  if (content_settings_type() != ContentSettingsType::BRAVE_ETHEREUM)
+    return;
+
+  // Move added pending request from pending_requests_ to
+  // ethereum_pending_requests_, because otherwise Chromium will replace this
+  // pending_request with next sub-request because it does not expect multiple
+  // sub-requests in the same type for one RequestPermissions request.
+  auto pending_request = pending_requests_.find(id.ToString());
+  DCHECK(pending_request != pending_requests_.end());
+
+  auto it = ethereum_pending_requests_.find(id.ToString());
+  if (it == ethereum_pending_requests_.end()) {
+    it = ethereum_pending_requests_
+             .insert(std::make_pair(
+                 id.ToString(), std::make_unique<EthereumPermissionRequests>()))
```

Comment:
can we change this class name to `GroupedPermissionRequests` or something more generic?

Reply:
https://github.com/brave/brave-core/pull/9443/commits/bfed18d3f531d2bbc98d69c7dd6a8bcc652cf0d3